lib/skate_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,27 @@
+defmodule SkateWeb.AuthController do
+  use SkateWeb, :controller
+  plug(Ueberauth)
+
+  alias SkateWeb.AuthManager
+  alias SkateWeb.Router.Helpers
+
+  def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
+    username = auth.uid
+    expiration = auth.credentials.expires_at
+
+    current_time = System.system_time(:second)
+
+    conn
+    |> Guardian.Plug.sign_in(
+      AuthManager,
+      username,
+      %{},
+      ttl: {expiration - current_time, :seconds}
```

Comment:
Just FYI this is one hour. That was all we needed initially on signs_ui, since mostly people pop it open to check on something, make a change, and then they're done. It's not meant to be left open for extended periods of time. After one hour they have to sign in again. Someone is currently working on getting the Oauth refresh_token flow up and running, though, to prevent that.